src/openrct2/ride/Ride.cpp

```diff
@@ -3472,13 +3471,24 @@ static void ride_shop_connected(Ride* ride)
     x = coordinates.x;
     y = coordinates.y;
 
-    tileElement = map_get_first_element_at(x, y);
+    TileElement* tileElement = map_get_first_element_at(x, y);
+    if (tileElement == nullptr)
+    {
+        return;
+    }
+
     do
     {
         if (tileElement->GetType() == TILE_ELEMENT_TYPE_TRACK && tileElement->AsTrack()->GetRideIndex() == ride->id)
```

Comment:
Break this up into two if statements much easier to read. I would introduce a bool found and then check that after the loop

Reply:
Ha the good old trick from the game actions huh?

Reply:
Yeah

Reply:
I did it slightly different which avoids two further calls to "AsTrack()" by simply having a TrackElement pointer being assigned or not.